server/src/utils.ts

```diff
@@ -72,17 +72,30 @@ export let findBscNativeOfFile = (
 
 // TODO: this doesn't handle file:/// scheme
 export let findNodeBuildOfProjectRoot = (
-  projectRootPath: p.DocumentUri
+  projectRootPath: p.DocumentUri,
+  pathToBinFromConfig: p.DocumentUri
 ): null | { buildPath: p.DocumentUri; isReScript: boolean } => {
-  let rescriptNodePath = path.join(projectRootPath, c.rescriptNodePartialPath);
-  let bsbNodePath = path.join(projectRootPath, c.bsbNodePartialPath);
+  if (pathToBinFromConfig !== "") {
+    if (fs.existsSync(pathToBinFromConfig)) {
+      if (pathToBinFromConfig.endsWith(c.rescriptBinName)) {
+        return { buildPath: pathToBinFromConfig, isReScript: true };
+      }
+      if (pathToBinFromConfig.endsWith(c.rescriptBinName)) {
+        return { buildPath: pathToBinFromConfig, isReScript: false };
+      }
+    }
+    return null;
+  } else {
+    let rescriptNodePath = path.join(projectRootPath, c.rescriptNodePartialPath);
+    let bsbNodePath = path.join(projectRootPath, c.bsbNodePartialPath);
 
-  if (fs.existsSync(rescriptNodePath)) {
```

Comment:
How about putting formatting changes in a separate commit so one gets a clean commit history.

Reply:
Ops. I'll do it.